lib/2wp-utils.js

```diff
@@ -107,8 +107,11 @@ const createPegoutRequest = async (rskTxHelper, amountInRBTC, requestSize = 1) =
  */
 const isUtxoRegisteredInBridge = async (rskTxHelper, peginTxHash, expectedUxtosCount = 1) => {
     const bridgeState = await getBridgeState(rskTxHelper.getClient());
-    return bridgeState.activeFederationUtxos
-        .filter(utxo =>  utxo.btcTxHash === peginTxHash).length === expectedUxtosCount;
+    const foundUtxos = bridgeState.activeFederationUtxos.filter(utxo =>  utxo.btcTxHash === peginTxHash);
+    if(foundUtxos.length > 0 && foundUtxos.length !== expectedUxtosCount) {
+        throw new Error(`Found ${foundUtxos.length} utxo(s) with the pegin tx hash ${peginTxHash} in the bridge, but expected to find ${expectedUxtosCount} utxo(s).`);
```

Comment:
Throwing exception here so it doesn't fail silently. Otherwise it might be confusing for some tests.

Reply:
Not sure about this, the method is called `isUtxoRegisteredInBridge`. I would expect it to return true or false. Now it's true or exception.

I understand the possible confusion. Perhaps it could be solved with a rename `assertUTXOsRegisteredInBridge`? Do nothing or throw exception

Reply:
Not really. In fact, this function gets called many times and returns `false` when the utxo is not yet in the Bridge. Which is ok. It's the `ensurePeginIsRegistered` function that calls the `isUtxoRegisteredInBridge` function many times.

When we register a pegin btc transaction in the Bridge, it could register either 1 or multiple utxos sharing the same btc transaction hash. If we are expecting 2 utxos from a pegin tx with the same btc transaction hash to be registered in the Bridge, but only 1 is registered, is it really successfully registered? Obviously not.
In a scenario like that, we want to throw an Exception.

Reply:
It cannot be called `assertUTXOsRegisteredInBridge` because it would fail if it doesn't find the utxo in the Bridge. Which is not what we want. Because this function is called multiple times, until the uxto gets registered. Only after finding the utxo and the expected utxo count it will return true.